### ISSUE TITLE
Make it so animationplayer can 'compile' into a GDScript / code . . .

### ISSUE THREAD

**Issue by charlesmlamb** [godotengine/godot-proposals#3137]
### Describe the project you are working on

A 2.5D game, with barbarians, lore . . .

### Describe the problem or limitation you are having in your project

I had a crazy idea, normally one uses animationplayer for simple systems, at same time, one often wants to make it into code, so one can customize it, add random stuff, or so . . . I was thinking, if I made a say light, in animationplayer, that flickered, with only keyframes . . .

So, like . .
time=0, light value=1
time=1, light value=2
time=2, light value=1
time=3, light value=2 . .

Could on take that logic, and make it into code, or ' compile ' it to GDScript . . . Having that template, could allow one to turn animationplayer ' effects ' or, camera movements, etc . . into code, that could then add stuff, like jitter, random effects, to say a flickering torch . . .

I'm not sure how to do this, the real advantage of coding these same effects, is they bec. much more customizable, add random systems, or so . . . It's just an idea, wanted to hear what pros think and, if it is impossible . . .

It'd allow one to make 'demos', in an animationplayer fast, then make it into code, and add stuff there . . Sort-of, over-coming the limitation of animationplayer, or so . . . Thx . . .

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Well, like above, a way, maybe for beginners, to quickly go from a simple animationplayer, to more complex stuff, or turn every animationplayer, into raw code or, so . . . Thx . . .

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

This is the idea, not sure it's good <3 . . Add a button, to animationplayer, so one converts it into a simple gdscript or, code . . . And, in a way that make an optimized, or at least really good version of it, ' nice ' code . . .

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I can't do this, it'd take pros also, not sure it's a good idea . . .

### Is there a reason why this should be core and not an add-on in the asset library?

Well, I'd like my games to, be 100 % code, if possible . . Not sure it's a good idea, it's better to have more customizable stuff, or so . . . Esp. for really big animations, or complex bezier paths, being able to get a ' template ' or, easy start, might not sure save time and, help games be even more ' logical ' <3 . . . .

**Comment by Xrayez** [comment]
Have you considered using a [`Tween`](https://docs.godotengine.org/en/stable/classes/class_tween.html)? One of the main advantages of using a tween is that you can control it via code. But bear in mind that `Tween` is going to be completely rewritten in Godot 4.0: godotengine/godot#41794.

**Comment by Zireael07** [comment]
Speaking of, what about a animplayer <> tween conversion?

**Comment by SlugFiller** [comment]
This could absolutely be done in an addon. Just scan the animation properties, and rewrite them to assignments. Adding a button to the animation tab from an editor plugin is a challenge (because you need to navigate through the editor's node tree), but not impossible.

This also would not be a good idea in the general case. An AnimationPlayer node can store any static animations much more efficiently than code can be stored. It's easy enough to control playing or seeking through code. If you want to mess with the values, e.g. add random jitter to a certain property when animated, you can create a proxy object which receives instructions from the AnimationPlayer via a property with a setter, performs the necessary modifications to it (e.g. add random jitter), and then send it to the actual target.

Basically, there's no need to "expand" the AnimationPlayer's actions into code. If anything, it would make it harder to work with, not more convenient.
